File: tasks/translation/output/Qwen/QwQ-32B-AWQ/codenet/direct/inline/comment/direct/Java/C++/s081988820_commented.cpp
#include <iostream>
#include <algorithm>
using namespace std;

class Main {
public:
    void solve() {
        int H, W;
        cin >> H >> W;
        int c[10][10];
        for (int i = 0; i < 10; ++i) {
            for (int j = 0; j < 10; ++j) {
                cin >> c[i][j];
            }
        }
        int min_cost[10];
        for (int i = 0; i < 10; ++i) {
            min_cost[i] = c[i][1];
        }
        for (int tc = 0; tc < 10; ++tc) {
            for (int i = 0; i < 10; ++i) {
                for (int j = 0; j < 10; ++j) {
                    min_cost[i] = min(min_cost[i], c[i][j] + min_cost[j]);
                }
            }
        }
        int ans = 0;
        for (int h = 0; h < H; ++h) {
            for (int w = 0; w < W; ++w) {
                int A;
                cin >> A;
                if (A >= 0) {
                    ans += min_cost[A];
                }
            }
        }
        cout << ans << endl;
    }
};

int main() {
    Main main;
    main.solve();
    return 0;
}
// End of Code

File: tasks/translation/output/Qwen/Qwen2.5-Coder-32B-Instruct-AWQ/codenet/direct/comment/comment/direct/Java/Python/s081988820_commented.py
import sys

class Main:
    def __init__(self):
        self.input = sys.stdin.read().split()
        self.index = 0

    def next(self):
        result = self.input[self.index]
        self.index += 1
        return result

    def next_int(self):
        return int(self.next())

    def solve(self):
        H = self.next_int()
        W = self.next_int()
        c = [[self.next_int() for _ in range(10)] for _ in range(10)]
        min_cost = [c[i][1] for i in range(10)]
        for _ in range(10):
            for i in range(10):
                for j in range(10):
                    min_cost[i] = min(min_cost[i], c[i][j] + min_cost[j])
        ans = 0
        for _ in range(H * W):
            A = self.next_int()
            if A >= 0:
                ans += min_cost[A]
        print(ans)

if __name__ == "__main__":
    main = Main()
    main.solve()

# End of Code
